lib/DxilContainer/DxilContainerAssembler.cpp

```diff
@@ -1373,6 +1388,78 @@ class DxilRDATWriter : public DxilPartWriter {
     }
   }
 
+  void UpdatePayloadInfo(const DxilModule &DM) {
+      Module* module = DM.GetModule();
+      assert(module);
+      DataLayout DL(module);
+
+      const DxilTypeSystem& DTS = DM.GetTypeSystem();
+      int TypeID = 0;
+      int FieldID = 0;
+      for (const auto& entry : DTS.GetPayloadAnnotationMap()) {
+        const StructType* type = entry.first;
+        const auto& annotation = entry.second;
+
+        RuntimeDataPayloadTypeInfo info; 
+        info.ID = TypeID;
+        info.Name = m_pStringBufferPart->Insert(type->getName());
+        m_pPayloadTypeTable->Insert(info);
+
+        for (unsigned i = 0; i < annotation->GetNumFields(); ++i) {
+            Type* fieldType = type->getElementType(i);
+            uint32_t size = 0;
+            bool fieldIsStructType = false;
+
+            if (fieldType->isAggregateType()) {
+              if (fieldType->isStructTy()) {
+                  size = DL.getTypeAllocSize(fieldType);
+                  fieldIsStructType = true;
+              } else {
+                ArrayType *arrayType = cast<ArrayType>(fieldType);
+                fieldType = arrayType->getElementType();
+                size = arrayType->getNumElements();
+                if (fieldType->isStructTy()) {
+                    size *= DL.getTypeAllocSize(fieldType);
+                    fieldIsStructType = true;
+                }
+              }
+            } else if (fieldType->isVectorTy()) {
+              VectorType *vectorType = cast<VectorType>(fieldType);
+              fieldType = vectorType->getElementType();
+              size = vectorType->getNumElements();
+            }
+            
+            RuntimeDataPayloadFieldInfo fieldInfo; 
+
+            const auto& fieldAnnotation = annotation->GetFieldAnnotation(i);
+
+            fieldInfo.ID = FieldID; 
+            fieldInfo.StructTypeName = 0;
+            if (!fieldIsStructType)
+              fieldInfo.Type = static_cast<uint32_t>(
+                  fieldAnnotation.GetCompType().GetKind());
+            else {
+              fieldInfo.Type =
+                  0xffffffffu; // indicates a struct type in the payload
```

Comment:
No named constant?

Reply:
This is removed, i use CompType now everywhere and for structs it is set to Invalid(0)